sdks/uniswapx-sdk/src/builder/V3DutchOrderBuilder.ts

```diff
@@ -0,0 +1,339 @@
+    }
+    return this;
+  }
+
+  outputOverrides(outputOverrides: BigNumber[]): this {
+    if (!this.info.cosignerData) {
+      this.initializeCosignerData({ outputOverrides });
+    } else {
+      this.info.cosignerData.outputOverrides = outputOverrides;
+    }
+    return this;
+  }
+
+  deadline(deadline: number): this {
+    super.deadline(deadline);
+    return this;
+  }
+
+  swapper(swapper: string): this {
+    super.swapper(swapper);
+    return this;
+  }
+
+  nonce(nonce: BigNumber): this {
+    super.nonce(nonce);
+    return this;
+  }
+
+  validation(info: ValidationInfo): this {
+    super.validation(info);
+    return this;
+  }
+
+  cosignerData(cosignerData: V3CosignerData): this {
+    this.decayStartBlock(cosignerData.decayStartBlock);
+    this.exclusiveFiller(cosignerData.exclusiveFiller);
+    this.exclusivityOverrideBps(cosignerData.exclusivityOverrideBps);
+    this.inputOverride(cosignerData.inputOverride);
+    this.outputOverrides(cosignerData.outputOverrides);
+    return this;
+  }
+
+  exclusiveFiller(exclusiveFiller: string): this {
+    if (!this.info.cosignerData) {
+      this.initializeCosignerData({ exclusiveFiller });
+    } else {
+      this.info.cosignerData.exclusiveFiller = exclusiveFiller;
+    }
+    return this;
+  }
+
+  exclusivityOverrideBps(exclusivityOverrideBps: BigNumber): this {
+    if (!this.info.cosignerData) {
+      this.initializeCosignerData({ exclusivityOverrideBps });
+    } else {
+      this.info.cosignerData.exclusivityOverrideBps = exclusivityOverrideBps;
+    }
+    return this;
+  }
+
+  // ensures that we only change non fee outputs
+  nonFeeRecipient(newRecipient: string, feeRecipient?: string): this {
+    invariant(
+      newRecipient !== feeRecipient,
+      `newRecipient must be different from feeRecipient: ${newRecipient}`
+    );
+    if (!this.info.outputs) {
+      return this;
+    }
+    this.info.outputs = this.info.outputs.map((output) => {
+      // if fee output then pass through
+      if (
+        feeRecipient &&
+        output.recipient.toLowerCase() === feeRecipient.toLowerCase()
+      ) {
+        return output;
+      }
+
+      return {
+        ...output,
+        recipient: newRecipient,
+      };
+    });
+    return this;
+  }
+
+  buildPartial(): UnsignedV3DutchOrder {
+    //build an unsigned order
+    this.checkUnsignedInvariants(this.info);
+    return new UnsignedV3DutchOrder(
+      Object.assign(this.getOrderInfo(), {
+        input: this.info.input,
+        outputs: this.info.outputs,
+        cosigner: this.info.cosigner,
+        startingBaseFee: this.info.startingBaseFee,
+      }),
+      this.chainId,
+      this.permit2Address
+    );
+  }
+
+  // A helper function for users of the class to easily the value to pass to maxAmount in an input
+  static getMaxAmountOut(
+    startAmount: BigNumber,
+    relativeAmounts: bigint[]
+  ): BigNumber {
+    if (relativeAmounts.length == 0) {
```

Comment:
I think we should support the empty curve for limit orders.

Reply:
Let's add a test for this too

Reply:
Fixed